src/Application.php

```diff
@@ -253,34 +254,38 @@ public function renderException($e, $output)
      */
     protected function doRunCommand(Command $command, InputInterface $input, OutputInterface $output)
     {
-        if ('core:configure' !== $this->getCommandName($input)
-            && 'core:update' !== $this->getCommandName($input)
-        ) {
+        $commandName = $this->getCommandName($input);
+
+        if ('core:configure' !== $commandName && 'core:update' !== $commandName) {
             if (null === $this->config) {
                 $this->config = Factory::createConfig();
             }
 
-            if (null === $this->adapter) {
-                if (null === $adapter = $this->config->get('adapter')) {
-                    $adapter = $this->determineAdapter();
-                }
-
-                $this->buildAdapter($adapter);
-            }
-
-            if (null === $this->issueTracker) {
-                $issueTracker = $this->config->get('issue_tracker');
+            if ('core:init' !== $commandName) {
```

Comment:
I'm not fully happy with this but I have some idea's to improve this in a follow-up PR.
Adapters should only build when they are requested (not on command init).